saml2/saml2-service-provider/src/main/java/org/springframework/security/saml2/provider/service/registration/OpenSamlMetadataAssertingPartyDetailsConverter.java

```diff
@@ -71,13 +71,16 @@ Collection<RelyingPartyRegistration.AssertingPartyDetails.Builder> convert(Input
 		if (xmlObject instanceof EntitiesDescriptor) {
 			EntitiesDescriptor descriptors = (EntitiesDescriptor) xmlObject;
 			for (EntityDescriptor descriptor : descriptors.getEntityDescriptors()) {
-				builders.add(convert(descriptor));
+				if (descriptor.getIDPSSODescriptor(SAMLConstants.SAML20P_NS) != null) {
+					builders.add(convert(descriptor));
+				}
 			}
 			return builders;
 		}
 		if (xmlObject instanceof EntityDescriptor) {
 			EntityDescriptor descriptor = (EntityDescriptor) xmlObject;
-			return Arrays.asList(convert(descriptor));
+			return descriptor.getIDPSSODescriptor(SAMLConstants.SAML20P_NS) == null ?
```

Comment:
I think this changes the semantics of the method, which we shouldn't do in a maintenance release. It will also break `RelyingPartyRegistrations#fromMetadata` since it relies on the return value being non-empty. I'd recommend this check be removed so that the original exception is still thrown.